crates/driver/src/domain/competition/solution/scoring.rs

```diff
@@ -108,12 +109,13 @@ impl Trade {
                 // to avoid loss of precision because we work with integers we first multiply
                 // and then divide:
                 // buy_amount = surplus * buy_price / sell_price
-                let surplus_in_buy_tokens: eth::U256 = surplus_in_surplus_token
-                    .full_mul(self.signed_buy.amount.0)
-                    .checked_div(self.signed_sell.amount.0.into())
-                    .ok_or(Error::Math(Math::DivisionByZero))?
-                    .try_into()
-                    .map_err(|_| Error::Math(Math::Overflow))?;
+                let surplus_in_buy_tokens = surplus_in_surplus_token
+                    .widening_mul(self.signed_buy.amount.0)
+                    .checked_div(alloy::primitives::U512::from(self.signed_sell.amount.0))
+                    .ok_or(Error::Math(Math::DivisionByZero))?;
+                let surplus_in_buy_tokens: eth::U256 =
```

Comment:
nit: this should not be needed since you already use `eth::U256::from_()` in the assignment.
```suggestion
                let surplus_in_buy_tokens =
```